cmd/kubelet/kubelet.go

```diff
@@ -82,23 +144,41 @@ func main() {
 	// short-circuit on verflag
 	verflag.PrintAndExitIfRequested()
 
-	// TODO(mtaufen): won't need this this once dynamic config is GA
-	// set feature gates so we can check if dynamic config is enabled
-	if err := utilfeature.DefaultFeatureGate.SetFromMap(defaultConfig.FeatureGates); err != nil {
-		die(err)
-	}
-	// validate the initial KubeletFlags, to make sure the dynamic-config-related flags aren't used unless the feature gate is on
-	if err := options.ValidateKubeletFlags(kubeletFlags); err != nil {
-		die(err)
+	// load Kubelet config file if necessary
```

Comment:
this is pretty tough to follow, and bleeds flag info down into the helper functions, which shouldn't be necessary. I expected a main() flow more like this:

```
kubeletConfig := defaultConfig

if configFile := kubeletFlags.KubeletConfigFile.Value(); len(configFile) > 0 {
  kubeletConfig, err = loadConfig(configFile)
  if err != nil {
    die(err)
  }
}

if dynamicConfigDir := kubeletFlags.DynamicConfigDir.Value(); len(dynamicConfigDir) > 0 {
  kubeletConfig, kubeletConfigController, err = app.BootstrapKubeletConfigController(kubeletConfig, dynamicConfigDir)
  if err != nil {
    die(err)
  }
}
```

that makes the file load and the dynamic config load very easy to understand in the main flow